lib/model/content.dart

```diff
@@ -1028,21 +1053,46 @@ class _ZulipContentParser {
       final grandchild = child.nodes[0];
       if (grandchild is! dom.Element) return null;
       if (grandchild.localName != 'img') return null;
-      if (grandchild.className.isNotEmpty) return null;
-      return grandchild;
+      return (child, grandchild);
     }();
 
     final debugHtmlNode = kDebugMode ? divElement : null;
-    if (imgElement == null) {
+    if (elements == null) {
       return UnimplementedBlockContentNode(htmlNode: divElement);
     }
 
+    final (linkElement, imgElement) = elements;
+    if (imgElement.className == 'image-loading-placeholder') {
+      return ImageNode(
+        srcUrl: null,
+        thumbnailUrl: null,
+        loading: true,
+        debugHtmlNode: debugHtmlNode);
+    }
+
+    final href = linkElement.attributes['href'];
+    if (href == null) {
+      return UnimplementedBlockContentNode(htmlNode: divElement);
+    }
     final src = imgElement.attributes['src'];
     if (src == null) {
       return UnimplementedBlockContentNode(htmlNode: divElement);
     }
 
-    return ImageNode(srcUrl: src, debugHtmlNode: debugHtmlNode);
+    final String srcUrl;
+    final String? thumbnailUrl;
+    if (src.startsWith('/user_uploads/thumbnail/')) {
```

Comment:
When this condition is false, let's have a check that `src == href`, and return unimplemented if they're unequal.

I believe those were always the same when thumbnailing isn't involved; if there are other cases that exist, I'd like to learn about them, either in manual testing as we use the app or as part of #190.

Reply:
There seems to be multiple cases where `src != href`, forexample src urls starting with `/external_content/` and `https://uploads.zulipusercontent.net/`.
Should they be added one by explicitly (using `startsWith`)?

Reply:
> they be added one by one explicitly
Did this for now.

Reply:
Yep, that sounds good.

Oh, it looks like we don't have a test case that exercises the `/external_content/` case, though. Do you have an example? That'd be good to add.